Failed to build release package

**Is this a BUG REPORT or FEATURE REQUEST?**:

/kind bug

**Description**:

There's no .git info in release package, and got following error when try to build the binaries :

```
~/workspace/src/volcano.sh/volcano-0.2# make
mkdir -p _output/bin
fatal: not a git repository (or any of the parent directories): .git
mkdir -p _output/release
go build -ldflags " -X 'volcano.sh/volcano/pkg/version.GitSHA=`git rev-parse HEAD`' -X 'volcano.sh/volcano/pkg/version.Built=`date "+%Y-%m-%d %H:%M:%S"`' -X 'volcano.sh/volcano/pkg/version.Version=v0.2'" -o=_output/bin/vc-scheduler ./cmd/scheduler
fatal: not a git repository (or any of the parent directories): .git
fatal: not a git repository (or any of the parent directories): .git
cmd/scheduler/main.go:33:2: cannot find package "volcano.sh/volcano/cmd/scheduler/app" in any of:
        /root/workspace/src/volcano.sh/volcano-0.2/vendor/volcano.sh/volcano/cmd/scheduler/app (vendor tree)
        /usr/local/go/src/volcano.sh/volcano/cmd/scheduler/app (from $GOROOT)
        /root/workspace/src/volcano.sh/volcano/cmd/scheduler/app (from $GOPATH)
cmd/scheduler/main.go:34:2: cannot find package "volcano.sh/volcano/cmd/scheduler/app/options" in any of:
        /root/workspace/src/volcano.sh/volcano-0.2/vendor/volcano.sh/volcano/cmd/scheduler/app/options (vendor tree)
        /usr/local/go/src/volcano.sh/volcano/cmd/scheduler/app/options (from $GOROOT)
        /root/workspace/src/volcano.sh/volcano/cmd/scheduler/app/options (from $GOPATH)
cmd/scheduler/main.go:37:2: cannot find package "volcano.sh/volcano/pkg/scheduler/actions" in any of:
        /root/workspace/src/volcano.sh/volcano-0.2/vendor/volcano.sh/volcano/pkg/scheduler/actions (vendor tree)
        /usr/local/go/src/volcano.sh/volcano/pkg/scheduler/actions (from $GOROOT)
        /root/workspace/src/volcano.sh/volcano/pkg/scheduler/actions (from $GOPATH)
cmd/scheduler/main.go:38:2: cannot find package "volcano.sh/volcano/pkg/scheduler/plugins" in any of:
        /root/workspace/src/volcano.sh/volcano-0.2/vendor/volcano.sh/volcano/pkg/scheduler/plugins (vendor tree)
        /usr/local/go/src/volcano.sh/volcano/pkg/scheduler/plugins (from $GOROOT)
        /root/workspace/src/volcano.sh/volcano/pkg/scheduler/plugins (from $GOPATH)
cmd/scheduler/main.go:41:2: cannot find package "volcano.sh/volcano/pkg/scheduler/util/assert" in any of:
        /root/workspace/src/volcano.sh/volcano-0.2/vendor/volcano.sh/volcano/pkg/scheduler/util/assert (vendor tree)
        /usr/local/go/src/volcano.sh/volcano/pkg/scheduler/util/assert (from $GOROOT)
        /root/workspace/src/volcano.sh/volcano/pkg/scheduler/util/assert (from $GOPATH)
Makefile:39: recipe for target 'vc-scheduler' failed
make: *** [vc-scheduler] Error 1
~/workspace/src/volcano.sh/volcano-0.2#

```

--- a/.travis.yml
+++ b/.travis.yml
@@ -51,7 +51,9 @@ jobs:
         - echo "publish release to github & dockerhub"
       deploy:
         - provider: script
-          script: make TAG=${TRAVIS_TAG} RELEASE_VER=${TRAVIS_TAG} release
+          script:
+            - make TAG=${TRAVIS_TAG} RELEASE_VER=${TRAVIS_TAG} release-env
+            - make TAG=${TRAVIS_TAG} RELEASE_VER=${TRAVIS_TAG} release
           on:
             tags: true
         - provider: releases

--- a/Makefile
+++ b/Makefile
@@ -17,15 +17,9 @@ RELEASE_DIR=_output/release
 REL_OSARCH=linux/amd64
 REPO_PATH=volcano.sh/volcano
 IMAGE_PREFIX=volcanosh/vc
-# If tag not explicitly set in users default to the git sha.
-TAG ?= $(shell git rev-parse --verify HEAD)
 RELEASE_VER=v0.2
-GitSHA=`git rev-parse HEAD`
-Date=`date "+%Y-%m-%d %H:%M:%S"`
-LD_FLAGS=" \
-    -X '${REPO_PATH}/pkg/version.GitSHA=${GitSHA}' \
-    -X '${REPO_PATH}/pkg/version.Built=${Date}'   \
-    -X '${REPO_PATH}/pkg/version.Version=${RELEASE_VER}'"
+
+include Makefile.def
 
 .EXPORT_ALL_VARIABLES:
 
@@ -77,6 +71,11 @@ e2e-test-kind:
 generate-yaml: init
 	./hack/generate-yaml.sh
 
+release-env:
+	./hack/build-env.sh release
+
+dev-env:
+	./hack/build-env.sh dev
 
 release: images generate-yaml
 	./hack/publish.sh

--- a/Makefile.def
+++ b/Makefile.def
@@ -1,0 +1,10 @@
+
+# If tag not explicitly set in users default to the git sha.
+TAG ?= $(shell git rev-parse --verify HEAD)
+GitSHA=`git rev-parse HEAD`
+Date=`date "+%Y-%m-%d %H:%M:%S"`
+LD_FLAGS=" \
+    -X '${REPO_PATH}/pkg/version.GitSHA=${GitSHA}' \
+    -X '${REPO_PATH}/pkg/version.Built=${Date}'   \
+    -X '${REPO_PATH}/pkg/version.Version=${RELEASE_VER}'"
+

--- a/defs/Makefile.dev.def
+++ b/defs/Makefile.dev.def
@@ -1,0 +1,10 @@
+
+# If tag not explicitly set in users default to the git sha.
+TAG ?= $(shell git rev-parse --verify HEAD)
+GitSHA=`git rev-parse HEAD`
+Date=`date "+%Y-%m-%d %H:%M:%S"`
+LD_FLAGS=" \
+    -X '${REPO_PATH}/pkg/version.GitSHA=${GitSHA}' \
+    -X '${REPO_PATH}/pkg/version.Built=${Date}'   \
+    -X '${REPO_PATH}/pkg/version.Version=${RELEASE_VER}'"
+

--- a/defs/Makefile.release.def
+++ b/defs/Makefile.release.def
@@ -1,0 +1,10 @@
+
+# If tag not explicitly set in users default to the git sha.
+TAG=__release_ver__
+GitSHA=__git_sha__
+Date=`date "+%Y-%m-%d %H:%M:%S"`
+LD_FLAGS=" \
+    -X '${REPO_PATH}/pkg/version.GitSHA=${GitSHA}' \
+    -X '${REPO_PATH}/pkg/version.Built=${Date}'   \
+    -X '${REPO_PATH}/pkg/version.Version=${RELEASE_VER}'"
+

--- a/hack/build-env.sh
+++ b/hack/build-env.sh
@@ -1,0 +1,10 @@
+#!/bin/bash
+
+ENV=$1
+
+GitSHA=`git rev-parse HEAD`
+
+cp defs/Makefile.${ENV}.def Makefile.def
+
+sed -i "s/__git_sha__/"${GitSHA}"/g" Makefile.def
+sed -i "s/__release_ver__/"${RELEASE_VER}"/g" Makefile.def

